File: app/src/main/java/it/unical/mat/embasp/specializations/dlv/desktop/DLVDesktopService.java
package it.unical.mat.embasp.specializations.dlv.desktop;

import it.unical.mat.embasp.base.Output;
import it.unical.mat.embasp.platforms.desktop.DesktopService;
import it.unical.mat.embasp.specializations.dlv.DLVAnswerSets;

/** DLV Solver for Desktop platforms */

public class DLVDesktopService extends DesktopService {

	public DLVDesktopService(final String exe_path) {
		super(exe_path);
		load_from_STDIN_option = "--";
	}

	@Override
	protected Output getOutput(final String output, final String error) {
		return new DLVAnswerSets(output, error);
	}

}

File: CSharp version/it/unical/mat/embasp/specializations/dlv/desktop/DLVDesktopService.cs
namespace it.unical.mat.embasp.specializations.dlv.desktop
{
    using Output = it.unical.mat.embasp.@base.Output;
    using DesktopService = it.unical.mat.embasp.platforms.desktop.DesktopService;

    public class DLVDesktopService : DesktopService
    {
        public DLVDesktopService(string exe_path) : base(exe_path) => load_from_STDIN_option = "--";

        protected internal override Output GetOutput(string output, string error) => new DLVAnswerSets(output, error);
    }
}